loopy/transform/iname.py

```diff
@@ -2504,18 +2504,23 @@ def does_insn_involve_iname(kernel, insn, *args):
 
 @for_each_kernel
 def rename_iname(kernel, old_iname, new_iname, existing_ok=False,
-                 within=None, preserve_tags=True):
-    """
+                 within=None, preserve_tags=True,
+                 raise_on_domain_mismatch: bool = __debug__):
```

Comment:
To avoid inconsistency, make the default `None`, and have a `if raise_on_domain_mismatch:` bit in `rename_inames` to centralize the default logic.